sugoi-api-core/src/main/java/fr/insee/sugoi/core/service/impl/PermissionServiceImpl.java

```diff
@@ -49,182 +48,133 @@ public class PermissionServiceImpl implements PermissionService {
+  public List<String> getAppManagerRoles(List<String> roles) {
+    return getStringRightsFromRegexList(roles, applicationManagerRoleList);
   }
 
-  private List<String> getUserRightList(SugoiUser sugoiUser, List<String> regexpListToSearch) {
-    List<String> searchRoleList =
-        getSearchRoleList(
-                sugoiUser,
-                "(?<realm>.*)",
-                "(?<userStorage>.*)",
-                "(?<application>.*)",
-                regexpListToSearch)
-            .stream()
-            .map(String::toUpperCase)
-            .collect(Collectors.toList());
-    List<String> roles =
-        sugoiUser.getRoles().stream()
-            .map(String::toUpperCase)
+  private List<String> getAllowedAttributePattern(
+      List<String> roles, String realm, String storage, String pattern) {
+    return getRightsFromRegexList(roles, applicationManagerRoleList).stream()
+        .map(appRight -> getPatternFromRegex(pattern, realm, storage, appRight.getApplication()))
+        .map(Pattern::pattern)
+        .collect(Collectors.toList());
+  }
+
+  private static List<String> getStringRightsFromRegexList(
+      List<String> roles, List<String> applicationManagerRoleList) {
+    return getRightsFromRegexList(roles, applicationManagerRoleList).stream()
+        .map(Right::toString)
+        .collect(Collectors.toList());
+  }
+
+  private static List<Right> getRightsFromRegexList(List<String> roles, List<String> regexpList) {
+    List<Pattern> validPatterns =
+        regexpList.stream()
             .map(
-                role -> {
-                  for (String searchRole : searchRoleList) {
-                    Pattern p = Pattern.compile(searchRole);
-                    Matcher m = p.matcher(role);
-                    if (m.matches()) {
-                      String realm = "";
-                      String userStorage = "";
-                      String application = "";
-                      try {
-                        realm = m.group("REALM");
-                      } catch (Exception e) {
-                      }
-                      try {
-                        userStorage = "_" + m.group("USERSTORAGE");
-                      } catch (Exception e) {
-                      }
-                      try {
-                        if (m.group("APPLICATION") != null) {
-                          if (realm.equals("")) {
-                            realm = "*";
-                            userStorage = "_*";
-                          }
-                          application = "\\" + m.group("APPLICATION");
-                        }
-                      } catch (Exception e) {
-                      }
-                      String res = realm + userStorage + application;
-                      return res;
-                    }
-                  }
-                  return null;
-                })
-            .filter(Objects::nonNull)
+                regexp ->
+                    getPatternFromRegex(
```

Comment:
Ok, but please explain a little (other method or comment) this usage of provided patterns to create new patterns for regex checkout any value